src/robotFactory.js

```diff
@@ -1,10 +1,77 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, {
+    x = 0,
+    y = 0,
+  }, chipVersion) {
+    this.name = name;
+    this.weight = weight;
 
-class FlyingRobot {}
+    this.coords = {
+      x,
+      y,
+    };
 
-class DeliveryDrone {}
+    this.chipVersion = chipVersion;
+  }
+
+  goForward(step = 1) {
+    this.coords.y = this.coords.y + step;
+  }
+
+  goBack(step = 1) {
+    this.coords.y = this.coords.y - step;
+  }
+  goLeft(step = 1) {
+    this.coords.x = this.coords.x - step;
+  }
+
+  goRight(step = 1) {
+    this.coords.x = this.coords.x + step;
+  }
+
+  getInfo() {
+    return `Robot: ${this.name}, Chip version: ${this.chipVersion}, `
+    + `Weight: ${this.weight}`;
+  }
+}
+
+class FlyingRobot extends BaseRobot {
+  constructor(name, weight, coords, chipVersion) {
+    super(name, weight, coords, chipVersion);
+
+    if (this.coords.x) {
```

Comment:
I think you wanted to check coords.z 😄 
```suggestion
    if (this.coords.z) {
```